pkg/clusterresource/builder.go

```diff
@@ -419,17 +419,21 @@ func (o *Builder) generateInstallConfigSecret() (*corev1.Secret, error) {
 	// Remove metadataService field from machinepool platform within installconfig.
 	// TODO: Remove this once https://bugzilla.redhat.com/show_bug.cgi?id=2098299 has been addressed.
 	if installConfig.Platform.AWS != nil {
-		ops := yamlpatch.Patch{
-			yamlpatch.Operation{
-				Op:   "remove",
-				Path: yamlpatch.OpPath("/compute/0/platform/aws/metadataService"),
-			},
-			yamlpatch.Operation{
-				Op:   "remove",
-				Path: yamlpatch.OpPath("/controlPlane/platform/aws/metadataService"),
-			},
+		d, err = yaml.YAMLToJSON(d)
+		if err != nil {
+			return nil, errors.Wrap(err, "error converting install-config.yaml to JSON")
+		}
+		d, err = sjson.DeleteBytes(d, `compute.0.platform.aws.metadataService`)
+		if err != nil {
+			return nil, errors.Wrap(err, "error removing metadataService field from install-config.yaml")
+		}
+		d, err = sjson.DeleteBytes(d, `controlPlane.platform.aws.metadataService`)
+		if err != nil {
+			return nil, errors.Wrap(err, "error removing metadataService field from install-config.yaml")
```

Comment:
These error messages should be distinguishable so we know which path broke.
```suggestion
			return nil, errors.Wrap(err, "error removing metadataService field from compute section of install-config.yaml")
		}
		d, err = sjson.DeleteBytes(d, `controlPlane.platform.aws.metadataService`)
		if err != nil {
			return nil, errors.Wrap(err, "error removing metadataService field from controlPlane section of install-config.yaml")
```